packages/components/src/Form/Inputs/InputFilters/InputFilters.tsx

```diff
@@ -106,26 +123,27 @@ const InputFiltersLayout: FC<InputFiltersProps> = ({
         const editFilter = () => setFieldEditing(filter.field)
 
         const handleDelete = () =>
-          setAssignedFilters(
-            [...assignedFilters].filter(
-              (assignedFilter) => assignedFilter.field !== filter.field
+          onChange(
+            filters.map((currentFilter) =>
+              currentFilter.field !== filter.field
+                ? currentFilter
+                : omit(currentFilter, 'value')
             )
           )
 
         const setFieldEditingValue = (value: string) => {
-          const filterIndex = assignedFilters.findIndex(
+          const filterIndex = filters.findIndex(
             (filter) => filter.field === fieldEditing
           )
 
-          const newAssignedFilters = [...assignedFilters]
-          const updateFilter = { ...newAssignedFilters[filterIndex], value }
-          newAssignedFilters[filterIndex] = updateFilter
+          const newFilters = [...filters]
```

Comment:
Last funky filter order fix:
```suggestion
          const newFilters = [...assignedFilters, ...unassignedFilters]
```